pkg/conditions/conditions_types.go

```diff
@@ -0,0 +1,79 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package conditions
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Conditions defines a list of Condition
+type Conditions []Condition
+
+// Type used for defining the conditiont Type field flavour
+type Type string
+
+const (
+	// Succeeded specifies that the resource has finished.
+	// For resources that run to completion.
+	Succeeded Type = "Succeeded"
+)
+
+// Condition defines the required fields for populating
+// Build controllers Conditions
+type Condition struct {
+	// Type of condition
+	// +required
+	Type Type `json:"type" description:"type of status condition"`
+
+	// Status of the condition, one of True, False, Unknown.
+	// +required
+	Status corev1.ConditionStatus `json:"status" description:"status of the condition, one of True, False, Unknown"`
+
+	// LastTransitionTime last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`
```

Comment:
LastTransitionTime should also be optional because at the beginning, there is no LastTransitionTime.
```
// +optional
```

Reply:
Thanks, added.